lxc/list.go

```diff
@@ -230,5 +221,96 @@ func (c *listCmd) run(config *lxd.Config, args []string) error {
 		}
 	}
 
-	return listContainers(cts, filters, len(cts) == 1)
+	columns_map := map[rune]Column{
+		'n': Column{i18n.G("NAME"), nameColumnData, true},
+		's': Column{i18n.G("STATE"), statusColumnData, true},
+		'4': Column{i18n.G("IPV4"), IP4ColumnData, false},
+		'6': Column{i18n.G("IPV6"), IP6ColumnData, false},
+		'e': Column{i18n.G("EPHEMERAL"), isEphemeralColumnData, false},
+		'S': Column{i18n.G("SNAPSHOTS"), numberSnapshotsColumnData, false},
+		'p': Column{i18n.G("PID"), PIDColumnData, false},
+	}
+
+	columns := []Column{}
+	for _, columnRune := range chosenColumnRunes {
+		if column, ok := columns_map[columnRune]; ok {
+			columns = append(columns, column)
+		} else {
+			fmt.Printf("%s does contain invalid column characters\n", chosenColumnRunes)
+			return nil
```

Comment:
I think you want to return an error here; the top level should print out the error, and then it will also exit(1), indicating failure.